src/storage.rs

```diff
@@ -186,6 +186,34 @@ impl Database {
         Ok(Database { db: Arc::new(db) })
     }
 
+    /// Opens the database at the given path in read-only mode.
+    ///
+    /// Read-only instances are optimized for query workloads and do not affect
+    /// write performance of the primary instance.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// * The database path does not exist or is not accessible
+    /// * The database cannot be opened in read-only mode
+    /// * Column families cannot be accessed
+    pub fn open_read_only(path: &Path, db_options: &DbOptions) -> Result<Database> {
```

Comment:
The `open()` and `open_read_only()` functions are identical except for one line of code.
Rather than writing the `open_read_only()` function as a separate function, it would be better to modify the existing `open()` function by adding the read_only parameter.